src/builtins.ts

```diff
@@ -113,11 +114,13 @@ function any(x: collections.Iterable): boolean {
 }
 
 function bool(x) {
-    if (x === undefined)
-        return false;
-    if (x['bool'] !== undefined)
-        return x.bool();
-    return x != 0;
+
+    if(x instanceof Object && x.asBool){
+        return x.asBool();
+    }
```

Comment:
(Nit) Space between if and condition, and condition and `{` (e.g. `if () {`).

Also, `} else {` on same line.